src/main/java/com/stripe/model/Source.java

```diff
@@ -1,228 +1,878 @@
-   * customer object.
+   * The `type` of the source. The `type` is a payment method, one of `ach_credit_transfer`,
+   * `ach_debit`, `alipay`, `bancontact`, `card`, `card_present`, `eps`, `giropay`, `ideal`,
+   * `multibanco`, `p24`, `paper_check`, `sepa_credit_transfer`, `sepa_debit`, `sofort`,
+   * `three_d_secure`, or `wechat`. An additional hash is included on the source with a name
+   * matching this value. It contains additional information specific to the [payment
+   * method](https://stripe.com/docs/sources) used.
    */
-  @Override
-  public Source delete(RequestOptions options) throws StripeException {
-    throw new InvalidRequestException(
-        "Source objects cannot be deleted. If you want to detach the source from a customer "
-        + "object, use detach().",
-         null, null, null, 0, null);
-  }
-  // </editor-fold>
+  @SerializedName("type")
+  String type;
 
-  // <editor-fold desc="detach">
   /**
-   * Detach a source.
+   * Either `reusable` or `single_use`. Whether this source should be reusable or not. Some source
+   * types may or may not be reusable by construction, while others may leave the option at
+   * creation. If an incompatible value is passed, an error will be returned.
    */
+  @SerializedName("usage")
+  String usage;
+
+  @SerializedName("wechat")
+  Wechat wechat;
+
+  /** Delete a specified source for a given customer. */
   public Source detach() throws StripeException {
-    return detach(null, null);
+    return detach((Map<String, Object>) null, (RequestOptions) null);
   }
 
-  /**
-   * Detach a source.
-   */
+  /** Delete a specified source for a given customer. */
   public Source detach(Map<String, Object> params) throws StripeException {
-    return detach(params, null);
+    return detach(params, (RequestOptions) null);
   }
 
-  /**
-   * Detach a source.
-   */
+  /** Delete a specified source for a given customer. */
   public Source detach(Map<String, Object> params, RequestOptions options) throws StripeException {
+    String url;
     if (this.getCustomer() != null) {
-      String url = String.format("%s/%s/sources/%s", classUrl(Customer.class), this.getCustomer(),
-          this.getId());
-      return request(RequestMethod.DELETE, url, params, Source.class, options);
+      url =
+          String.format(
+              "%s%s",
+              Stripe.getApiBase(),
+              String.format("/v1/customers/%s/sources/%s", this.getCustomer(), this.getId()));
     } else {
       throw new InvalidRequestException(
-          "This source object does not appear to be currently attached to a customer object.",
-          null, null, null, 0, null);
+          "Unable to construct url because [customer] field(s) are all null",
+          null,
+          null,
+          null,
+          0,
+          null);
     }
+    return request(ApiResource.RequestMethod.DELETE, url, params, Source.class, options);
```

Comment:
18) Delete (custom): Instance method to delete-like a sub-resource on itself, relying on ID attributes of the instance, and its reference to parent resource. POST should be used instead. Currently one method has this behavior.

Reply:
https://github.com/stripe/stripe-java/blob/2aa5db5c39dc6e91228507798083f373904cc8c0/src/main/java/com/stripe/model/Source.java#L187-L206